internal/v2/s2av2.go

```diff
@@ -307,12 +307,37 @@ func removeServerNamePort(serverName string) string {
 	return name
 }
 
-func createStream(ctx context.Context, s2av2Address string) (s2av2pb.S2AService_SetUpSessionClient, error) {
+type s2AGrpcStream struct {
+	stream s2av2pb.S2AService_SetUpSessionClient
+}
+
+func (x *s2AGrpcStream) Send(m *s2av2pb.SessionReq) error {
+	return x.stream.Send(m)
+}
+
+func (x *s2AGrpcStream) Recv() (*s2av2pb.SessionResp, error) {
+	return x.stream.Recv()
+}
+
+func (x *s2AGrpcStream) CloseSend() error {
+	return x.stream.CloseSend()
+}
+
+func createStream(getS2AStream func(ctx context.Context, s2av2Address string) (stream.S2AStream, error), ctx context.Context, s2av2Address string) (stream.S2AStream, error) {
```

Comment:
Can we add unit tests that catch the changes here?

Reply:
Isn't this an inner function and should be left untested?

I searched in `s2av2_test.go` and `s2av2_e2e_test.go` and didn't find any direct test on this function.

Reply:
Previously, this function was tested indirectly because we're creating real streams to the fake S2Av2 server.

However, now, we're not testing the changes to this function at all. So I think it is reasonable to add unit tests for this function. This should be fine to as long as the test is in the same package as the `s2av2.go` file.

Reply:
Done.